### ISSUE TITLE
[COMPATIBILITY]: Legion 5 17ACH6H Custom Fan Control

### ISSUE THREAD

**Issue by Ytimes2** [BartoszCichecki/LenovoLegionToolkit#440]
### Version

2.7.0

### OS

Windows 11 22H2 Build 22621.819

### Device

Legion 5 17ACH6H

### BIOS version

GKCN54WW

### Which features of the application did you test?

Legion 5 17ACH6H does not have the option of fan control, hopefully it will be compatible!

### Which features of the application do not work as expected?

Legion 5 17ACH6H does not have the option of fan control, hopefully it will be compatible!

**Comment by BartoszCichecki** [comment]
Fan control is not supported on gen 6 devices (except for max speed which is already available).

**Comment by Ytimes2** [comment]
> 

How do I set it to max speed...

> Fan control is not supported on gen 6 devices (except for max speed which is already available).

How do I set it to max speed...

**Comment by BartoszCichecki** [comment]
You need to be in Custom power mode.

**Comment by Ytimes2** [comment]
> You need to be in Custom power mode. And a "Settings" icon shows up next to the dropdown and you can set it there.

I remember there is a software called "legionfancontrol" that can adjust the fan, but it feels weird to use it

**Comment by BartoszCichecki** [comment]
Yes there is. It manipulates the embedded controller directly into setting different fan speeds. This is neither secure nor 100% safe. This app uses only the methods that Vantage or LegionZone uses, hence no fan control on gen 6 devices.

**Comment by Ytimes2** [comment]
> Yes there is. It manipulates the embedded controller directly into setting different fan speeds. This is neither secure nor 100% safe. This app uses only the methods that Vantage or LegionZone uses, hence no fan control on gen 6 devices.

alright! thanks!

**Comment by InvaderDolan** [comment]
> Yes there is. It manipulates the embedded controller directly into setting different fan speeds. This is neither secure nor 100% safe. This app uses only the methods that Vantage or LegionZone uses, hence no fan control on gen 6 devices.

What are the default RPMs for Legion 7 Gen 6 for example? It think if there is any need to keep LFC?

**Comment by BartoszCichecki** [comment]
@InvaderDolan I don't know. If you join the Discord that is linked in readme and ask, probably someone knows. I didn't go as far as extracting the info directly from EC firmware, but I know fan tables are stored there.
